tachyon/crypto/commitments/polynomial_openings.h

```diff
@@ -16,54 +16,54 @@
 namespace tachyon::crypto {
 
 // A single polynomial with a single opening.
-template <typename Poly>
+template <typename Poly, typename PolyOracle = Poly>
 struct PolynomialOpening {
   using Field = typename Poly::Field;
   using Point = typename Poly::Point;
 
-  // P₀
-  base::DeepRef<const Poly> poly;
+  // polynomial -> P₀ || commitment -> C₀
+  base::DeepRef<const PolyOracle> poly_oracle;
   // x₀
   base::DeepRef<const Point> point;
   // P₀(x₀)
   Field opening;
 
   PolynomialOpening() = default;
-  PolynomialOpening(base::DeepRef<const Poly> poly,
+  PolynomialOpening(base::DeepRef<const PolyOracle> poly_oracle,
                     base::DeepRef<const Point> point, Field&& opening)
-      : poly(poly), point(point), opening(std::move(opening)) {}
+      : poly_oracle(poly_oracle), point(point), opening(std::move(opening)) {}
 };
 
 // A single polynomial with multi openings.
-template <typename Poly>
+template <typename Poly, typename PolyOracle = Poly>
 struct PolynomialOpenings {
   using Field = typename Poly::Field;
 
-  // P₀
-  base::DeepRef<const Poly> poly;
+  // polynomial -> P₀ || commitment -> C₀
+  base::DeepRef<const PolyOracle> poly_oracle;
   // [P₀(x₀), P₀(x₁), P₀(x₂)]
   std::vector<Field> openings;
 
   PolynomialOpenings() = default;
-  PolynomialOpenings(base::DeepRef<const Poly> poly,
+  PolynomialOpenings(base::DeepRef<const PolyOracle> poly_oracle,
                      std::vector<Field>&& openings)
-      : poly(poly), openings(std::move(openings)) {}
+      : poly_oracle(poly_oracle), openings(std::move(openings)) {}
 };
 
 // Multi polynomials with multi openings grouped by shared points.
-template <typename Poly>
+template <typename Poly, typename PolyOracle = Poly>
```

Comment:
Please modify the comment to encompass both cases when `PolyOracle` is 'Poly' and when it's 'Commitment'.